client/state/data-layer/wpcom/sites/comments/index.js

```diff
@@ -264,6 +266,71 @@ export const announceEditFailure = ( { dispatch }, action ) => {
 	);
 };
 
+const bulkChangeCommentStatus = action => {
+	const { comments, siteId, status } = action;
+	const commentIds = map( comments, 'commentId' );
+	return http(
+		{
+			method: 'POST',
+			path: `/sites/${ siteId }/comments/status`,
+			apiVersion: '1',
+			body: {
+				comment_ids: commentIds,
+				status,
+			},
+		},
+		action
+	);
+};
+
+const handleBulkChangeCommentStatusSuccess = (
+	{ comments, siteId, status, refreshCommentListQuery },
+	{ results }
+) => dispatch => {
+	if ( !! refreshCommentListQuery ) {
+		dispatch( requestCommentsList( refreshCommentListQuery ) );
+	}
+
+	dispatch( removeNotice( 'comment-notice-bulk-error' ) );
+
+	const updatedComments = filter( comments, ( { commentId } ) => includes( results, commentId ) );
+	forEach( updatedComments, ( { commentId, postId } ) =>
+		dispatch( bypassDataLayer( changeCommentStatusAction( siteId, postId, commentId, status ) ) )
+	);
+
+	const message = get(
+		{
+			approved: translate( 'All selected comments approved.' ),
+			unapproved: translate( 'All selected comments unapproved.' ),
+			pending: translate( 'All selected comments unapproved.' ),
+			spam: translate( 'All selected comments marked as spam.' ),
+			trash: translate( 'All selected comments moved to trash.' ),
+		},
+		status
+	);
+	dispatch(
+		successNotice( message, {
+			id: 'comment-notice-bulk',
+			isPersistent: true,
+		} )
+	);
+};
```

Comment:
why is this a thunk? I think we can just return a list of actions to dispatch instead.

```js
// no need to translate all the messages on every call when only one is needed
const getMessage = status => {
	switch ( status ) {
		case 'approved': return translate( '…' );
		case 'unapproved': return translate( '…' );
		…
		default:
			blowUp();
	}
}

…

if ( !! refreshcommentListQuery ) {
	return requestCommentsList( refreshCommentListQuery );
}

const updatedComments = comments
	.filter( ( { commentId } ) => results.includes( commentId ) )
	.map( ( { commentId, postId } ) => bypassDataLayer( … ) );

return [
	removeNotice( 'comment-notice-bulk-error' ),
	...updatedComments,
	successNotice( getMessage( status ) ), {
		id: 'comment-notice-bulk',
		isPersistent: true,
	} )
]
```